packages/x-date-pickers/src/DatePicker/shared.tsx

```diff
@@ -103,7 +90,7 @@ export function useDatePickerDefaultizedProps<TDate, Props extends BaseDatePicke
     };
   }, [themeProps.localeText]);
 
-  const slots = themeProps.slots ?? uncapitalizeObjectKeys(themeProps.components);
+  const slots = themeProps.slots;
```

Comment:
Nitpick: It probably no longer makes sense to have a separate variable for this. 🤔 
```suggestion
```